test/query.test.js

```diff
@@ -195,4 +196,16 @@ describe('parquetQuery', () => {
       { a: 'abc', b: 5, c: 2, d: true, e: [1, 2] },
     ])
   })
+
+  it('reads data efficiently with filter', async () => {
+    const file = countingBuffer(await asyncBufferFromFile('test/files/page_indexed.parquet'))
+    const rows = await parquetQuery({ file, filter: { quality: 'good' }, rowStart: 1, rowEnd: 5 })
+    expect(rows).toEqual([
+      { row: 10n, quality: 'good' },
+      { row: 29n, quality: 'good' },
+      { row: 32n, quality: 'good' },
+      { row: 37n, quality: 'good' },
+    ])
+    expect(file.fetches).toBe(2)
```

Comment:
maybe add a comment to understand that without the optimization, we would have fetched 3 times? Or add another test, with a higher rowEnd, to show that the number of fetches depends on the number of required rows